oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyStateFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.document;
+
+import org.apache.jackrabbit.oak.api.PropertyState;
+import org.apache.jackrabbit.oak.commons.Compression;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DocumentPropertyStateFactory {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DocumentPropertyStateFactory.class);
+
+    public static PropertyState createPropertyState(DocumentNodeStore store, String name, String value, Compression compression) {
+
+        if (compression != null && !compression.equals(Compression.NONE) && value.length() > CompressedDocumentPropertyState.getCompressionThreshold()) {
```

Comment:
the length check here would result to true for `-1` - shouldn't there be a test for that, did that not fail for some reason?

Reply:
The tests are checking default threshold exceeded or not, it didnt failed because of passing Compression.NONE as argument

Reply:
There should be a test that should fail now though. The condition is currently wrong. 

Probably we should also use GZIP by default, otherwise the feature cannot be enabled at all?

Reply:
Done. We can enable the feature by calling differently factory method.

Reply:
> We can enable the feature by calling differently factory method.

But that would require a code change. What I was referring to is enabling the feature by configuration - eg via the system property. We should have that - otherwise the code is not useable and disable in a hard-coded way. I do think we should use GZIP in the default, not NONE.

Reply:
Done.

Reply:
The test still doesn't fail consistently. I'm referring to `DocumentPropertyStateFactoryTest.createPropertyStateWithDefaultCompression`. That is one test at least that should fail with the current code. The reason it doesn't fail is that it depends on test execution order. Some other test method in that class is executed first (eg `createPropertyStateWithCompressionThresholdNotExceeded`) and that sets the threshold, so that `createPropertyStateWithDefaultCompression` then runs on false assumptions.

Things that require fixing:
* createPropertyStateWithDefaultCompression should check that it actually uses the default - i.e. it should have an assertEquals before even executing the actual property creation (and that would currently fail, depending on test execution order)
* in addition to such asserts in each method, there must also be a `@After` and a `@Before` that resets the threshold
* plus the actual code fix

Reply:
Done